examples/cross_dataset.py

```diff
@@ -0,0 +1,157 @@
+"""Cross-dataset motor imagery classification example.
+
+This example demonstrates how to perform cross-dataset evaluation using MOABB,
+training on one dataset and testing on another.
+"""
+
+# Standard library imports
+import logging
+from typing import Any, List
+
+import matplotlib.pyplot as plt
+
+# Third-party imports
+import mne
+import numpy as np
+import pandas as pd
+from pyriemann.estimation import Covariances
+from pyriemann.spatialfilters import CSP
+from sklearn.pipeline import Pipeline
+from sklearn.preprocessing import FunctionTransformer
+from sklearn.svm import SVC
+
+# MOABB imports
+from moabb import set_log_level
+from moabb.analysis.plotting import score_plot
+from moabb.datasets import BNCI2014001, Zhou2016
+from moabb.evaluations.evaluations import CrossDatasetEvaluation
+from moabb.paradigms import MotorImagery
+
+
+# Configure logging
+set_log_level("WARNING")
+logging.getLogger("mne").setLevel(logging.ERROR)
+
+
+def get_common_channels(datasets: List[Any]) -> List[str]:
```

Comment:
There is a `match_all` method in base paradigm:
https://github.com/NeuroTechX/moabb/blob/357cd121ff1a27c610f005cb44ff632085032628/moabb/paradigms/base.py#L429

Reply:
please use this method @ali-sehar